controllers/ciskubebenchreport/ciskubebenchreport_controller.go

```diff
@@ -0,0 +1,349 @@
+			secValues := valuesForSection(s, LabelsForGroup(labelGroupSectionSummary))
+
+			secValues["report_name"] = reportValues["report_name"]
+			secValues["node_name"] = reportValues["node_name"]
+			secValues["status"] = status
+
+			//Expose the metric.
+			BenchmarkSectionSummary.With(
+				secValues,
+			).Set(count)
+		}
+
+	}
+}
+
+func publishResultMetrics(report *aqua.CISKubeBenchReport, targetLabels []ReportLabel) {
+
+	reportValues := valuesForReport(report, targetLabels)
+
+	// Add node name to section metrics
+	for _, s := range report.Report.Sections {
+		secValues := valuesForSection(s, targetLabels)
+
+		for _, t := range s.Tests {
+			// Add node name and node type to result metrics
+			for _, r := range t.Results {
+				resValues := valuesForResult(r, targetLabels)
+
+				// Set report_name label from report
+				resValues["report_name"] = reportValues["report_name"]
+
+				// Inherit labels from report and section only if they are enabled
+				if _, found := reportValues["node_name"]; found {
+					resValues["node_name"] = reportValues["node_name"]
+				}
+				if _, found := secValues["node_type"]; found {
+					resValues["node_type"] = secValues["node_type"]
+				}
+				if _, found := secValues["section_name"]; found {
+					resValues["section_name"] = secValues["section_name"]
+				}
+
+				//Expose the metric.
+				BenchmarkResultInfo.With(
+					resValues,
+				).Set(1)
+			}
+		}
+	}
+
+}
+
+func valuesForReport(report *aqua.CISKubeBenchReport, labels []ReportLabel) map[string]string {
+	result := map[string]string{}
+	for _, label := range labels {
+		if label.Scope == FieldScopeReport {
+			result[label.Name] = reportValueFor(label.Name, report)
+		}
+	}
+	return result
+}
+
+func valuesForSection(sec aqua.CISKubeBenchSection, labels []ReportLabel) map[string]string {
+	result := map[string]string{}
+	for _, label := range labels {
+		if label.Scope == FieldScopeSection {
+			result[label.Name] = secValueFor(label.Name, sec)
+		}
+	}
+	return result
+}
+
+func valuesForResult(res aqua.CISKubeBenchResult, labels []ReportLabel) map[string]string {
+	result := map[string]string{}
+	for _, label := range labels {
+		if label.Scope == FieldScopeResult {
+			result[label.Name] = resValueFor(label.Name, res)
+		}
+	}
+	return result
+}
+
+func resValueFor(field string, res aqua.CISKubeBenchResult) string {
+	switch field {
+	case "test_number":
+		return res.TestNumber
+	case "test_desc":
+		return res.TestDesc
+	case "test_status":
+		return res.Status
+	default:
+		// Error?
+		return ""
+	}
+}
+
+func secValueFor(field string, sec aqua.CISKubeBenchSection) string {
+	switch field {
+	case "node_type":
+		return sec.NodeType
+	case "section_name":
+		return sec.Text
+	default:
+		// Error?
+		return ""
+	}
+}
+
+func reportValueFor(field string, report *aqua.CISKubeBenchReport) string {
+	switch field {
+	case "report_name":
+		return report.Name
+	case "node_name":
+		return report.Name
```

Comment:
Do you think there could be cases where the report name is different than the node name? I see starboard adds a label containing the resource name as well but I don't know under what circumstances they would be different. I'm wondering if it would be safe to drop `report_name` and just use `node_name` for CIS benchmarks. WDYT? Too risky?

Reply:
I don't see a situation where the report_name would be other than the node name. I checked the `starboard.resource.name` and seems to be the same as the `node_name` too.

I will refactor it so `node_name` is used instead.